holidays/countries/azerbaijan.py

```diff
@@ -26,54 +27,52 @@ class Azerbaijan(HolidayBase):
     country = "AZ"
 
     def _populate(self, year: int) -> None:
-        def _add_observed(hol_date: date, hol_name: str) -> None:
-            next_workday = hol_date + td(days=+1)
-            while next_workday.year == year and (
-                self._is_weekend(next_workday) or self.get(next_workday)
-            ):
-                next_workday += td(days=+1)
-            _add_holiday(next_workday, f"{hol_name} (Observed)")
-
-        def _add_holiday(hol_date: date, hol_name: str) -> None:
-            if hol_date.year == year:
-                self[hol_date] = hol_name
+        def add_to_list(hol_date: date, hol_name: str) -> None:
```

Comment:
I believe this will be refactored to a class level method later (as a general shift to a cleaner OOP approach) combined with `_add_holiday()`.  As for now just:
```suggestion
        def collect_holiday(hol_date: date, hol_name: str) -> None:
```

Reply:
Done.